frontend/public/components/sidebars/resource-sidebar-samples.tsx

```diff
@@ -0,0 +1,297 @@
+      },
+      {
+        title: 'Allow reading a ConfigMap in a specific namespace (for RoleBinding)',
+        description:
+          'This "Role" is allowed to read a "ConfigMap" named "my-config" (must be bound with a "RoleBinding" to limit to a single "ConfigMap" in a single namespace).',
+        id: 'read-configmap-within-ns',
+        kind: referenceForModel(RoleModel),
+      },
+      ...clusterRoleBindingSamples,
+    ],
+  )
+  .setIn([referenceForModel(ClusterRoleModel)], clusterRoleBindingSamples);
+
+export const getResourceSidebarSamples = (kindObj: K8sKind, yamlSamplesList: FirehoseResult) => {
+  const yamlSamplesData = !_.isEmpty(yamlSamplesList)
+    ? _.filter(
+        yamlSamplesList.data,
+        (sample: K8sResourceKind) =>
+          sample.spec.targetResource.apiVersion === apiVersionForModel(kindObj) &&
+          sample.spec.targetResource.kind === kindObj.kind,
+      )
+    : [];
+  const existingSamples = defaultSamples.get(referenceForModel(kindObj)) || [];
+  const extensionSamples = !_.isEmpty(yamlSamplesData)
+    ? yamlSamplesData.map((sample: K8sResourceKind) => {
+        return {
+          id: sample.metadata.uid,
+          ...sample.spec,
+        };
+      })
+    : [];
+  return [...existingSamples, ...extensionSamples];
+};
+
+const ResourceSidebarSample: React.FC<ResourceSidebarSampleProps> = ({
+  sample,
+  loadSampleYaml,
+  downloadSampleYaml,
+}) => {
+  const { highlightText, title, img, description, id, yaml } = sample;
+  const kind = sample.targetResource ? sample.targetResource.kind : sample.kind;
+  return (
+    <li className="co-resource-sidebar-item">
+      <h3 className="h4">
+        <span className="text-uppercase">{highlightText}</span> {title}
+      </h3>
+      {img && <img src={img} className="co-resource-sidebar-item__img img-responsive" />}
+      <p>{description}</p>
+      <Button type="button" variant="link" isInline onClick={() => loadSampleYaml(id, yaml, kind)}>
+        <PasteIcon className="co-icon-space-r" />
+        Try it
+      </Button>
+      <Button
+        type="button"
+        variant="link"
+        isInline
+        className="pull-right"
+        onClick={() => downloadSampleYaml(id, yaml, kind)}
+      >
+        <DownloadIcon className="co-icon-space-r" />
+        Download YAML
+      </Button>
+    </li>
+  );
+};
+
+export const ResourceSidebarSamples: React.FC<ResourceSidebarSamplesProps> = ({
+  samples,
+  loadSampleYaml,
+  downloadSampleYaml,
+}) => {
+  return (
+    <ol className="co-resource-sidebar-list">
+      {_.map(samples, (sample) => (
+        <ResourceSidebarSample
+          key={sample.id}
+          sample={sample}
+          loadSampleYaml={loadSampleYaml}
+          downloadSampleYaml={downloadSampleYaml}
+        />
+      ))}
+    </ol>
+  );
+};
+
+type Sample = {
+  highlightText: string;
+  title: string;
+  img: string;
+  description: string;
+  id: string;
+  yaml: string;
+  kind?: K8sKind;
```

Comment:
Is this really a `K8sKind` (the model) or a string?

I'd probably have one way to specify this instead of both `kind` and `targetResource.kind`

Reply:
Updated so both existing and extension samples use `targetResource.kind`, a string.